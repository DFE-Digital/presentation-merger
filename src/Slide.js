export default class Slide {
  constructor(content, style) {
    this.originalContent = content;
    this.originalStyle = style;
  }

  get content() {
    return JSON.parse(JSON.stringify(this.originalContent));
  }

  get style() {
    return JSON.parse(JSON.stringify(this.originalContent));
  }
}
