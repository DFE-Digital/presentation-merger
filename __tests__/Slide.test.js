import Slide from "../src/Slide";

describe("Slide", () => {
  describe(".content", () => {
    it("copies without reference the originalContent", () => {
      let originalContent = {
        greet: "hello world",
        deep: { thing: "original" }
      };

      const slide = new Slide(originalContent);
      let actual = slide.content;

      expect(actual).toEqual(originalContent);

      actual.greet = "new world";
      actual.deep.thing = "changed";
      expect(actual).toHaveProperty("greet", "new world");
      expect(originalContent).toHaveProperty("greet", "hello world");

      expect(actual).toHaveProperty("deep.thing", "changed");
      expect(originalContent).toHaveProperty("deep.thing", "original");
    });
  });

  describe(".style", () => {
    it("copies without reference the originalContent", () => {
      let originalContent = {
        greet: "hello world",
        deep: { thing: "original" }
      };

      const slide = new Slide(originalContent);
      let actual = slide.style;

      expect(actual).toEqual(originalContent);

      actual.greet = "new world";
      actual.deep.thing = "changed";
      expect(actual).toHaveProperty("greet", "new world");
      expect(originalContent).toHaveProperty("greet", "hello world");

      expect(actual).toHaveProperty("deep.thing", "changed");
      expect(originalContent).toHaveProperty("deep.thing", "original");
    });
  });
});
