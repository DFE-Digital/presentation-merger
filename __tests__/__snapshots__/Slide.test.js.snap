// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide .styleIDs returns the ids for the slide 1`] = `
Array [
  "dp1",
]
`;

exports[`Slide .styles includes the styles for the given slide 1`] = `
Array [
  Object {
    "style:drawing-page-properties": Object {
      "presentation:background-objects-visible": "true",
      "presentation:background-visible": "true",
      "presentation:display-date-time": "true",
      "presentation:display-footer": "true",
      "presentation:display-page-number": "false",
    },
    "style:family": "drawing-page",
    "style:name": "dp1",
  },
]
`;
